src/index.tsx

```diff
@@ -5,26 +5,16 @@ import * as serviceWorker from "./serviceWorker"
 import { BrowserRouter as Router, Switch, Route } from "react-router-dom"
 import { Provider, useSelector } from "react-redux"
 import { I18nProvider } from "@lingui/react"
-import { createStore, applyMiddleware } from "redux"
-import createSagaMiddleware from "redux-saga"
-import rootReducer from "./reducers/rootReducer"
-import rootSaga from "./sagas/rootSaga"
-import { LocaleReducerState } from "./reducers/localeReducer"
+import { LocaleState } from "./reducers/localeSlice"
 import { Spin } from "antd"
+import store from "./store"
+import { AppState } from "./reducers/rootReducer"
 
 // Lazy loading main pages
 const Home = lazy(() => import("./pages/Home/index"))
 
-const sagaMiddleware = createSagaMiddleware()
-
-const store = createStore(rootReducer, applyMiddleware(sagaMiddleware))
-
-sagaMiddleware.run(rootSaga)
-
 const I18nWrapper: React.FC = () => {
-  const { locale } = useSelector<any, LocaleReducerState>(
-    store => store.localeReducer,
-  )
+  const { locale } = useSelector<AppState, LocaleState>(store => store.locale)
```

Comment:
There's a few issues with this selector usage:

- Semantically, the argument isn't the `store`.  It's the `state` that's inside that store.  Strongly recommend changing the variable name to `state` instead.
- You generally shouldn't manually declare the generics up front.  Instead, declare it as `const someValue = useSelector( (state: AppState) => state.slice.someValue)`.  The return type will be inferred automatically.
- By returning a larger object and then destructuring the result, the component will re-render whenever that larger value is updated.  This is not usually what you want - you should select the smallest possible piece of state this component depends on, like: `const locale = useSelector( (state: AppState) => state.locale.locale)`.

Reply:
Fixed all remarks.